components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/store/IdentityStoreImpl.java

```diff
@@ -108,46 +99,45 @@ public User getUser(String username) throws IdentityStoreException, UserNotFound
     }
 
     @Override
-    public User getUser(Claim claim) throws IdentityStoreException, UserNotFoundException {
+    public User getUser(Claim claim) throws IdentityStoreException, UserNotFoundException, DomainException {
 
         UserNotFoundException userNotFoundException = new UserNotFoundException("User not found for the given name.");
 
         String attributeName = claim.getClaimURI(); // TODO: Get the attribute name from the claim manager.
         String attributeValue = claim.getValue();
 
-        Map<String, IdentityStoreConnector> identityStoreConnectorsMap = resolveDomain().getIdentityStoreConnectorMap();
+        Map<String, IdentityStoreConnector> identityStoreConnectorsMap = resolveDomain(claim)
+                .getIdentityStoreConnectorMap();
 
         for (IdentityStoreConnector identityStoreConnector : identityStoreConnectorsMap.values()) {
             try {
+                // The getUserBuilder method will set a unique user id
```

Comment:
getUserBuilder belongs to a connector. It only has the connector specific user Id. Unique userId should be set by the IdentityStore.